core/GlobalState.cc

```diff
@@ -1768,6 +1768,10 @@ void GlobalState::deleteMethodSymbol(MethodRef what) {
     ENFORCE(fnd != ownerMembers.end());
     ENFORCE(fnd->second == what);
     ownerMembers.erase(fnd);
+    for (const auto typeArgument : whatData->typeArguments()) {
+        this->typeArguments[typeArgument.id()] = this->typeArguments[0].deepCopy(*this);
+    }
+    // This drops the core::Method, which drops the ArgInfo's the method owned.
```

Comment:
```suggestion
    // This drops the existing core::Method, which drops the `ArgInfo`s the method owned.
```

The "existing" bit you can take or leave; one can argue that it's kind of redundant with the actual semantics of the assignment, but I do think it helps clarify what `core::Method` is being dropped here.

I think the quoting around the `ArgInfo` is an improvement regardless.